src/scenes/boot-scene.js

```diff
@@ -1,40 +1,62 @@
 import Ball from "src/models/ball";
+import  {SCREEN_WIDTH} from "src/constants";
+import  {SCREEN_HEIGHT} from "src/constants";
 
 export default class BootScene extends Phaser.Scene {
   constructor(){
     super({key: "Boot"});
+
+    this.width = SCREEN_WIDTH;
+    this.height = SCREEN_HEIGHT;
+    this.isGameOver = false;
   }
 
   preload() {
     this.load.image("red-ball", "assets/imgs/ball.png");
     this.load.image("tiles", "assets/tileMap/block_spritesheet.png");
     this.load.image("phaser-logo", "assets/imgs/logo.png");
+    this.load.image("gameOverImage", "assets/imgs/gameover.png");
+    this.load.image("diamond", "assets/imgs/diamond.png");
 
     this.load.tilemapTiledJSON("map", "assets/tileMap/blocks.json");
 
     this.load.audio("bounce", "assets/audio/bounce.wav");
-    this.load.audio("gameOver", "assets/audio/gameOver.wav");
-  }
-
-  update(time, delta){
-    this.redBall.update(time, delta);
-    if(this.redBall.isDead){
-      this.gameOverSound.play();
-      this.redBall.reset();
-    }
+    this.load.audio("gameOverSound", "assets/audio/gameOver.wav");
   }
 
   create() {
     this.groundLayer = this.addGround();
     this.redBall = new Ball(this, 200, 200);
-    this.gameOverSound = this.sound.add("gameOver");
+    this.diamonds = this.addAllDiamonds();
+    this.gameOverSound = this.sound.add("gameOverSound");
+    this.keyR = this.input.keyboard.addKey("R");
+    this.score = 0;
+    this.scoreImage = this.add.image(270,25, "diamond");
+    this.scoreImage.setScrollFactor(0);
+    this.scoreText = this.add.text(290, 20, "x " + this.score);
+    this.scoreText.setScrollFactor(0);
 
     this.cameras.main.setBounds(0, 0, this.groundLayer.width, this.groundLayer.height);
     this.cameras.main.startFollow(this.redBall, true);
 
     this.physics.add.collider(this.groundLayer, this.redBall);
   }
 
+  update(time, delta){
+    this.redBall.update(time, delta);
+
+    if(this.redBall.isDead && !this.isGameOver){
+      this.addGameOverMessage();
+      this.addGameOverSound();
+      this.addGameOver();
+      this.isGameOver = true;
+    }
+
+    if(this.redBall.isDead){
+      this.rToReset();
```

Comment:
what is a rToReset? please use a better name for this `resetGame` could be an option

Reply:
make the name's change to resetGame()